### ISSUE TITLE
riak exponential growth of riak connections

### ISSUE THREAD

**Issue by traviskirstine** [mapproxy/mapproxy#454]
Hi,

I'm currently using MapProxy v12 utilizing a riak cache configure as a gunicorn (v18.0) app.  Our backend riak cluster has multiple nodes behind HAProxy load balancer.  Our basic use case is that we have multiple TMS sources that are assemble into a single cache which is seeded on the fly based (WMTS).  It appears that MapProxy has issues releasing / creating connection caches to riak during this process, as the use client service increase riak will continue to create new connection caches until reaches the max connection as defined in the  HAProxy at which point the service become unresponsive as it is no longer able to cache or retrieve tiles.  I'm able to watch this increase by tracking the open files on the mapproxy server.

lsof | grep gunicorn | grep riak | wc -l
2954

I'm using MapProxy v1.10.1 with the exact same use case and gunicorn setup and not having this issue.
CentOS Linux release 7.7.1908 (Core)
Python 2.7.5 (default, Nov  6 2016, 00:28:07)

**Comment by traviskirstine** [comment]
It appears this issue is related to the riak python package leaking connections as described here: [https://github.com/basho/riak-python-client/issues/500](url)

Current workaround is to use riak v 2.2.0

**Comment by ramunasd** [comment]
This really might be riak-python issue.
We do not close connection explicitly, moreover we try to reuse it across threads.

**Comment by traviskirstine** [comment]
I agree more than likely an issue with riak-python package and the recommended version 2.4.2 seems to work without issue, v 2.7.0 seems to be the problem.  Unfortunately is does not appear that the riak-python-client project is active anymore so I don't think it will get addressed there,
